test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -626,32 +628,55 @@ public String toString() {
         return result;
     }
 
+    private static boolean matchWithRange(String nodeVersionString, List<VersionRange> acceptedVersionRanges, XContentLocation location) {
+        var unqualifiedNodeVersionString = nodeVersionString.replace("-SNAPSHOT", "");
+        try {
+            Version version = Version.fromString(unqualifiedNodeVersionString);
+            return acceptedVersionRanges.stream().anyMatch(v -> v.contains(version));
+        } catch (IllegalArgumentException e) {
+            throw new XContentParseException(
+                location,
+                "[version] range node selector expects a semantic version format (x.y.z), but found " + unqualifiedNodeVersionString,
+                e
+            );
+        }
+    }
+
     private static NodeSelector parseVersionSelector(XContentParser parser) throws IOException {
         if (false == parser.currentToken().isValue()) {
             throw new XContentParseException(parser.getTokenLocation(), "expected [version] to be a value");
         }
-        List<VersionRange> skipVersionRanges = parser.text().equals("current")
-            ? List.of(new VersionRange(Version.CURRENT, Version.CURRENT))
-            : SkipSection.parseVersionRanges(parser.text());
+
+        final Predicate<String> nodeMatcher;
+        final String versionSelectorString;
+        if (parser.text().equals("current")) {
+            var currentUnqualified = Build.current().version().replace("-SNAPSHOT", "");
+            nodeMatcher = nodeVersion -> currentUnqualified.equals(nodeVersion.replace("-SNAPSHOT", ""));
+            versionSelectorString = "current";
```

Comment:
Should this be set to `currentUnqualified` (or perhaps a string containing both, to make it clear that current was set, but that it was replaced with the actual current version)?

Reply:
Good suggestion